Library/Formula/telegram.rb

```diff
@@ -0,0 +1,40 @@
+class Telegram < Formula
+  homepage "https://github.com/vysheng/tg"
+  url "https://github.com/vysheng/tg/archive/1.0.5.1.tar.gz"
+  sha1 "72f6a9f89ee9db90b573bad25e567a2015115848"
+
+  depends_on "openssl"
+  depends_on "libevent"
+  depends_on "libconfig"
+  depends_on "readline"
+  depends_on "lua"
+
+  patch :DATA
+
+  def install
+    ENV.append "CFLAGS", "-I#{HOMEBREW_PREFIX}/include -I#{Formula["readline"].prefix}/include"
+    ENV.append "LDFLAGS", "-L#{HOMEBREW_PREFIX}/lib -L#{Formula["readline"].prefix}/lib"
+
+    system "./configure", "--prefix=#{prefix}"
+    system "make"
+
+    bin.install "bin/telegram-cli"
+    etc.install "tg-server.pub"
+  end
+
+  test do
+    `#{bin}/telegram-cli -h`.include? "telegram-cli Usage"
```

Comment:
Sorry, there's no `--help` available in this program :-(

```
~ ❯❯❯ telegram-cli --help
telegram-cli: illegal option -- -
telegram-cli Usage
  -u                  specify username (would not be asked during authorization)
  -k                  specify location of public key (possible multiple entries)
  -v                  increase verbosity (0-ERROR 1-WARNIN 2-NOTICE 3+-DEBUG-levels)
  -N                  message num mode
  -c                  config file name
  -p                  use specified profile
  -l                  log level
  -f                  during authorization fetch all messages since registration
  -E                  diable auto accept of encrypted chats
  -s                  lua script file
  -W                  send dialog_list query and wait for answer before reading input
  -C                  disable color output
  -R                  disable readline
  -d                  daemon mode
  -L <log-name>       log file name
  -U <user-name>      change uid after start
  -G <group-name>     change gid after start
  -D                  disable output
  -P <port>           port to listen for input commands
  -S <socket-name>    unix socket to create
```